src/main/java/com/xmslyz/luhn/logic/Computable.java

```diff
@@ -0,0 +1,6 @@
+package com.xmslyz.luhn.logic;
+
+public interface Computable {
```

Comment:
W sumie tak dochodzę do wniosku, że ta klasa to typowy serwis domenowy. Interfejs może nazwałbym tu LuhnChecksumService, a klasę implementującą LuhnChecksumServiceImpl (taka konwencja w Javie, implementacje mają przyrostek Impl)

Reply:
Nie chciałbym interfejsu wiązać z jednym konkretnym algorytmem. (Pomijam fakt nazwania projektu Algortymem Luhna 😅 ) 

Chciałbym tego kodu użyć niebawem do obliczania checksumy dla chociażby PESELU (a ten używa innych wag) stąd ta ogólna nazwa.

Reply:
Jasne. Teoretycznie w dalszym ciągu mamy do czynienia w obliczaniem jakiejś sumy kontrolnej, w takim wypadku `ChecksumSevice/ChecksumComputingService` jest bardziej ogólna, a poszczególne implementacje mogłyby się nazywać `LuhnChecksumComputingService` itp, ALE
taka konwencja narzuca kaganiec ograniczający możliwości zastosowanie - zwraca typ `String` i wymaga dokładnie jednego argumentu typu `String`. W przypadku algorytmu liczącego np. sumę kontrolną na postawie dwóch ciągów znaków wejściowych mielibyśmy problem.

To sprawdziłaby się, gdyby wynikiem tego projektu była np. aplikacja wykorzystująca tą sumę kontrolną, ale implementację dostarczał by dostarczał klient (interfejs narzucałby tu kontrakt). Wg mnie tutaj każdy typ kalkulacji to jednak osobny przypadek "biznesowy", zupełnie przypadkiem mogący (ale niekoniecznie!) mieć taką samą strukturę (przyjmuje i zwraca typ `String` i liczy jakąś sumę kontrolną :wink:)